util/bloom/bloom.go

```diff
@@ -116,6 +131,10 @@ func UnmarshalBinary(data []byte) (*Filter, error) {
 	}
 	copy(f.prefix[:], data[4:8])
 	f.data = data[8:]
+	f.preimageStagingBuffer = make([]byte, 0, 4+32)
+	f.hashStagingBuffer = make([]uint32, f.numHashes+3)
```

Comment:
I think we should try to avoid reallocating the array.
Why don't we use preimageStagingBuffer = preimageStagingBuffer[:0].

Reply:
I assume that you're referring to this line : `f.preimageStagingBuffer = make([]byte, 0, 4+32)` -
note that this function is creating a new Filter object, so there is no `reallocating`. it's the initial allocation.